Adding the Gradle JPI Plugin as a dependency breaks Jenkins Test Harness

### Jenkins and plugins versions report

n/a

### What Operating System are you using (both controller, and any agents involved in the problem)?

N/A

### Reproduction steps

N/A

### Expected Results

N/A

### Actual Results

N/A

### Anything else?

Hello! 

I've got a bit of a random issue i'm hoping you might be able to point me in the right direction on.
If this isn't the right medium to ask, _please_ shamelessly send me somewhere else :)

I'm working on a custom Gradle plugin related to the Jenkins plugin i maintain (the [Jenkins Templating Engine](https://github.com/jenkinsci/templating-engine-plugin)).

It extends this Gradle Plugin to automatically package a user's JTE-compatible pipeline libraries into a stand alone Jenkins Plugin they can install. 

for some reason, when i add `org.jenkins-ci.tools:gradle-jpi-plugin:0.43.0` as a dependency in my Gradle Plugin, it breaks the Jenkins Test Harness.

I get some weird issue with:

<details> <summary> Error Log </summary>

```
org.jvnet.hudson.reactor.ReactorException: java.lang.NoSuchMethodError: hudson.Extension.optional()Z
	at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:282)
	at jenkins.InitReactorRunner.run(InitReactorRunner.java:50)
	at jenkins.model.Jenkins.executeReactor(Jenkins.java:1164)
	at jenkins.model.Jenkins.<init>(Jenkins.java:964)
	at hudson.model.Hudson.<init>(Hudson.java:85)
	at org.jvnet.hudson.test.JenkinsRule.newHudson(JenkinsRule.java:688)
	at org.jvnet.hudson.test.JenkinsRule.before(JenkinsRule.java:404)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:595)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoSuchMethodError: hudson.Extension.optional()Z
	at hudson.ExtensionFinder$Sezpoz.logLevel(ExtensionFinder.java:741)
	at hudson.ExtensionFinder$Sezpoz._find(ExtensionFinder.java:716)
	at hudson.ExtensionFinder$Sezpoz.find(ExtensionFinder.java:695)
	at hudson.ClassicPluginStrategy.findComponents(ClassicPluginStrategy.java:346)
	at hudson.ExtensionList.load(ExtensionList.java:381)
	at hudson.ExtensionList.ensureLoaded(ExtensionList.java:317)
	at hudson.ExtensionList.getComponents(ExtensionList.java:183)
	at jenkins.model.Jenkins$6.onInitMilestoneAttained(Jenkins.java:1161)
	at jenkins.InitReactorRunner$1.onAttained(InitReactorRunner.java:85)
	at org.jvnet.hudson.reactor.ReactorListener$Aggregator.lambda$onAttained$3(ReactorListener.java:102)
	at org.jvnet.hudson.reactor.ReactorListener$Aggregator.run(ReactorListener.java:109)
	at org.jvnet.hudson.reactor.ReactorListener$Aggregator.onAttained(ReactorListener.java:102)
	at org.jvnet.hudson.reactor.Reactor$1.run(Reactor.java:177)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
```

</details>

commenting out this dependency, while breaking my plugin, allows the Jenkins Test Harness to run successfully again. 

so while this isn't a bug with the Gradle JPI Plugin, i'm hoping you might have a lead you could send me on. Really at a loss for what could be causing this one.

--- a/src/main/java/hudson/Extension.java
+++ b/src/main/java/hudson/Extension.java
@@ -33,5 +33,7 @@ import static jenkins.YesNoMaybe.MAYBE;
  * @author Kohsuke Kawaguchi
  */
 public @interface Extension {
+    double ordinal() default 0;
+    boolean optional() default false;
     YesNoMaybe dynamicLoadable() default MAYBE;
 }

